grammars/acme.cson

```diff
@@ -33,74 +33,99 @@ repository:
-            name: 'punctuation.definition.string.end.acme'
         name: 'string.quoted.double.assembly.acme'
       }
-      # absolut addressing/numbering
       {
-        match: '\\#(\'.\'|[^\\s\']+)'
+        begin: '\''
+        end: '\''
+        name: 'string.quoted.single.assembly.acme'
+      }
+      # absolut addressing/numbering (including lo/hibyte by < or >)
+      {
+        match: '\\#(\\>?\\<?\\(?\\$?\\w*\\)?|[^\\s]\\h?*)'
         name: 'constant.numeric.hex.acme'
       }
-      # hex, prefixed with dollar ($)
+      # hex, prefixed with dollar ($) or 0x
       {
-        match: '-?\\$[A-Fa-f0-9]+'
+        match: '\\$\\h+|0x\\h+'
         name: 'constant.numeric.hex.acme'
       }
       # binary prefixed with %
       {
-        match: '%[01]+'
+        match: '%[01]+|%[\\#\\.]+|0b[01]+|0b[\\#\\.]+'
         name: 'constant.numeric.binary.acme'
       }
-      # decimal
+#     float
+      {
+        match: '\\d*\\.\\d+\\b'
+        name: 'constant.numeric.float.acme'
+      }
+      # decimal (but not float)
       {
-        match: '\\b([0-9]+)\\b'
+        match: '\\b\\d+\\b'
         name: 'constant.numeric.decimal.acme'
       }
+#     octal
+      {
+        match: '&([0-7]+)\\b'
+        name: 'constant.numeric.octal.acme'
+      }
     ]
 
   # assembler directives
   directives:
     patterns: [
-      # File and Symbol control
+      # File and Symbol control (pseudos)
       {
-        match:  '!\\b(?i:source|src|eof|endoffile|realpc|pseudopc|zn|(sub)?zone|set|to|bin(ary)?|sl|symbollist|warn|error|serious|initmem|cbm|addr(ess)?)\\b'
+        match:  '\\!(?i:source|src|pseudopc|zn|zone|to|bin(ary)?|symbollist|sl|initmem)\\b'
         name:   'support.function.pseudo.acme'
       }
-      # Parsing control
+      # Parsing control (more pseudos)
       {
-        match:  '!\\b(?i:cpu|al|as|rl|rs|08|by|byte|([bl]e)?(16|24|32)|wo|word|h|hex|fi|fill|align|skip|ct|convtab|tx|text|pet|scr|scrxor|raw)\\b'
-        name:   'support.function.pseudo.acme'
-      }
-      # Macro control
-      {
-        match:  '(?i:!\\bmacro|\\+[a-zA-Z0-9])\\b'
+        match:  '\\!(?i:macro|cpu|al|as|rl|rs|8|by|byte|le16|be16|16|wo|word|le24|be24|24|le32|be32|32|hex|h|fill|fi|align|skip|ct|convtab|tx|text|pet|scr|scrxor|raw|address|addr)\\b'
         name:   'support.function.pseudo.acme'
       }
       # Conditional control
       {
-        match:  '!\\b(?i:if|if(n)?def|for|do|while|until)\\b'
+        match:  '\\!(?i:if|if(n)?def|for|do|set|warn|error|serious|eof|endoffile)\\b'
         name:   'keyword.control.conditional.acme'
       }
-      # Conditional control II
       {
-        match:  '\\b(?i:else|while|until)\\b'
+        match:  '(?i:while|until|else)\\b'
         name:   'keyword.control.conditional.acme'
       }
-      # Operators
-      {
-        match:  '!|\\+|\\-|\\/|\\*|<<|>>|&|\\||\\^|=|<|>|\\:|\\|\\|'
-        name:   'keyword.operator.acme'
-      }
-      # Operators II
+      # Operators (don't work)
```

Comment:
What is the trouble with this? When I uncomment it, it works fine, as do Operators II. 

Have you checked if your syntax theme in Atom highlights operators? I found that some syntax themes _simply do not highlight certain scopes_. I had that problem when writing a grammar for 6809 assembly, my grammar code was correct but my syntax theme simply didn't include code for highlighting certain scopes. Switching between several syntax themes sometimes reveals that problem.

EDIT: I use Monokai Seti as syntax theme

Reply:
I hadn't tried, I will uncomment them.